tests/integration/v3_watch_restore_test.go

```diff
@@ -69,7 +73,7 @@ func TestV3WatchRestoreSnapshotUnsync(t *testing.T) {
 		t.Fatal(errW)
 	}
 	if err := wStream.Send(&pb.WatchRequest{RequestUnion: &pb.WatchRequest_CreateRequest{
-		CreateRequest: &pb.WatchCreateRequest{Key: []byte("foo"), StartRevision: 5}}}); err != nil {
+		CreateRequest: &pb.WatchCreateRequest{Key: []byte("foo"), StartRevision: 25}}}); err != nil {
```

Comment:
Just curious, is watching from 25 revision counted as "old revision that were created in synced watcher group in the first place"?

Reply:
Thanks for the link! Not sure I understand it correctly: we don't defrag it so the revision is still available in kv store. Is it correct?

Reply:
Yeah, from my understanding, as long as the requested revision is not compacted, the key value is still available. 

<img width="993" alt="image" src="https://user-images.githubusercontent.com/54131596/230696907-d17f2e99-6501-4eb3-880c-c02a141100fd.png">

Here is a simple diagram. /cc @ahrtr @serathius in case the understanding or direction is wrong ==